docs/P4c-test-documentation.md

```diff
@@ -0,0 +1,143 @@
+# **Adding New Tests to the P4C Repository**
+
+**This document provides a guide for developers who want to add new tests to the P4C repository. It explains where to place test files, how to manage expected output files, and how to handle tests that are expected to fail (Xfail).**
+
+---
+
+## **1. Adding "Positive" Tests**
+
+Positive tests are those where the P4 compiler should process the input without any errors.
+
+### Steps for Adding Positive Tests
+
+1. **Source File Location**:
+
+   - Positive tests are located in multiple directories based on the purpose and the backend/tool being tested:
+
+     - `p4c/testdata/p4_16_samples`: General positive tests for the P4\_16 language.
+     - `p4c/testdata/p4_16_samples_outputs`: Expected outputs for these tests.
+     - Other directories may contain tests for specific backends or scenarios (e.g., BMv2 backend or PSA architecture).
+     - For BMv2-specific tests with packets, the source file name must match the pattern `*-bmv2.p4`.
+
+   - Review the structure of the `testdata` directory to understand the organization and differences.
+   - Ensure the file is named descriptively to reflect its purpose (e.g., `example_positive_test.p4`).
+
+2. **Expected Output Files**:
+
+   - Run the P4 compiler on the source file using the command:
+     ```bash
+     ./build/p4test path/to/source_file.p4
+     ```
+   - This generates expected output files.
+   - Place the generated expected output files in the appropriate directory, such as `p4c/testdata/p4_16_samples_outputs`.
+
+3. **Updating Expected Outputs**:
+
+   - If a change to the P4 compiler modifies the expected outputs, you can regenerate them by rerunning the tests with:
+     ```bash
+     ./build/p4test path/to/source_file.p4 --update-outputs
```

Comment:
I can find no mention of `update-outputs` anywhere in the p4c repo.  Where did you find such an option?